### ISSUE TITLE
WPS440 Found block variables overlap: false positive

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#793]
# Bug report

## What's wrong

Consider this code:

```python
from some import MIDDLEWARE

MIDDLEWARE = ('nplusone.ext.django.NPlusOneMiddleware',) + MIDDLEWARE
```

## How is that should be

This code explicitly uses the same name on both sides. We cannot refactor it to use `+=`, because the order is important.

I am not sure how to treat it.

**Comment by ea-volkov** [comment]
Hello, @sobolevn! I would like to work on this issue.

**Comment by sobolevn** [comment]
Thanks a lot, @ea-volkov ! 👍

**Comment by ea-volkov** [comment]
@sobolevn I need fix cases with one binary operation only?

**Comment by sobolevn** [comment]
Nope, with any name usage:
- `some = some.upper()`
- `some = [...] + some`
- `some = call(some)`
- etc

**Comment by sobolevn** [comment]
Sorry, I had some time to release new version today, so I have implemented this myself.

@ea-volkov thanks again for your time and effort! Check out new Github Actions!
